tasks/enable.yml

```
@@ -0,0 +1,104 @@
+  fail:
+    msg: Fapolicyd does not support trust setting fapolicyd_setup_trust
+  ignore_errors: true
+  when:
+    - fapolicyd_setup_trust | length > 0
+    - ansible_facts.distribution_version is version("8.2", "<=")
+  register: __failed_check_trust
+
+- name: Check integrity compatibility
+  fail:
+    msg: Fapolicyd does not support integrity setting fapolicyd_setup_integrity
+  ignore_errors: true
+  when:
+    - fapolicyd_setup_integrity | length > 0
+    - ansible_facts.distribution_version is version("8.3", "<=")
+  register: __failed_check_integrity
+
+- name: Check trust files compatibility
+  fail:
+    msg: >-
+      Fapolicyd does not support trust files setting fapolicyd_add_trusted_file
+  ignore_errors: true
+  when:
+    - fapolicyd_add_trusted_file | length > 0
+    - ansible_facts.distribution_version is version("8.3", "<=")
+  register: __failed_check_trusted_file
+
+- name: Check failed conditions
+  fail:
+    msg: Multiple failed conditions
+  # failed_when: true
+  when: __failed_check_trust is failed or __failed_check_integrity is failed  or
+    __failed_check_trusted_file is failed
+
+- name: Install fapolicyd packages
+  package:
+    name:
+      - fapolicyd
+    state: present
+
+- name: Install fapolicyd-selinux packages
+  package:
+    name:
+      - fapolicyd-selinux
+    state: present
+  when: ansible_facts.distribution_version is version("8.3", ">=")
+
+- name: Copy fapolicyd configuration file
+  template:
+    src: "{{ __fapolicyd_conf }}.j2"
+    dest: "{{ __fapolicyd_dir }}/{{ __fapolicyd_conf }}"
+    owner: root
+    group: fapolicyd
+    mode: '0644'
+
+- name: Run fapolicyd configuration check
+  command: fapolicyd-cli --check-config
+  check_mode: false
+  changed_when: false
+  when: ansible_facts.distribution_version is version("8.6", ">=")
+
+- name: Trustdb cleanup
+  command: fapolicyd-cli --file delete /
+  when: fapolicyd_add_trusted_file | length > 0
+  changed_when: true
+  failed_when: false
+
+- name: Add file to trustdb
+  command: fapolicyd-cli --file add {{ item | quote }}
+  loop: "{{ (fapolicyd_add_trusted_file is string) |
+    ternary([fapolicyd_add_trusted_file], fapolicyd_add_trusted_file) }}"
+  when: fapolicyd_add_trusted_file | length > 0
+  changed_when: true
+
+- name: Start fapolicyd service
+  service:
+    name: fapolicyd
+    state: restarted
+    enabled: true
+  ignore_errors: true
+  register: __fapolicyd_restart
+
+- name: Check fapolicyd logs
+  command: journalctl -n5 -u {{ __fapolicyd_services }}
+  register: __results
+  changed_when: false
+  when: __fapolicyd_restart is failed
+
+- name: Making sure fapolicyd does not run if it was set so
+  service:
+    name: fapolicyd
+    state: stopped
+    enabled: false
+  when: not fapolicyd_setup_enable_service
+
+- name: Print fapolicyd logs
+  debug:
+    msg: "{{ __results.stdout_lines }}"
+  failed_when: true
+  when:
+    - __fapolicyd_restart is failed
+    - __results.stdout_lines
```

I won't rely on Ansible to translate a non-empty output into `true`, let's rather check for the length of the output. Correct me if I am wrong.
```suggestion
    - __results.stdout_lines | length > 0
```

---

> It's just an existence check.

If there is a chance that `__results` or `__result.stdout_lines` could be undefined, then better to use `- __results.stdout_lines is defined`